qiskit/transpiler/__init__.py

```diff
@@ -34,84 +34,164 @@
 
-If you'd like to work directly with a
-preset pass manager you can use the :func:`~.generate_preset_pass_manager`
-function to easily generate one. For example:
+The function :func:`.generate_preset_pass_manager` creates the "preset pass managers".
+These are all instances of :class:`.PassManager`, so are used by pass a :class:`.QuantumCircuit` to
+the :meth:`.PassManager.run` method.  More specifically, the preset pass managers are instances of
+:class:`.StagedPassManager`, which allows greater configuration of the individual stages of a
+tranpsilation.
+
+A preset pass manager has up to six named stages.  These are summarized, in order of execution,
+below, with more in-depth information in the following subsections.
+
+``init``
+    Abstract-circuit optimizations, and reduction of multi-qubit operations to 1- and 2-qubit
+    operations.  See :ref:`transpiler-preset-stage-init` for more detail.
+
+``layout``
+    Choose an initial mapping of virtual qubits to physical qubits, including expansion of the
+    circuit to contain explicit ancillas.  This stage sometimes subsumes ``routing``.  See
+    :ref:`transpiler-preset-stage-layout` for more detail.
+
+``routing``
+    Insert gates into the circuit to ensure it matches connectivity constraints of the
+    :class:`.Target`.  The inserted gates need not match the target ISA yet, so are often just
+    ``swap`` instructions.  This stage is sometimes omitted, when the ``layout`` stage handles its
+    job.  See :ref:`transpiler-preset-stage-routing` for more detail.
+
+``translation``
+    Convert all gates in the circuit to ones matching the :class:`Target`\\ s ISA.  See
+    :ref:`transpiler-preset-stage-translation` for more detail.
+
+``optimization``
+    Low-level, hardware-aware optimizations.  Unlike the abstract optimizations of the ``init``
+    stage, this stage acts on a physical circuit.  See :ref:`transpiler-preset-stage-optimization`
+    for more detail.
+
+``scheduling``
+    Insert :class:`~.circuit.Delay` instructions to make the wall-clock timing of a circuit
+    explicit.  This may also include hardware-aware online error reduction techniques such as
+    dynamical decoupling, which are dependent on knowing wall-clock timings.  See
+    :ref:`transpiler-preset-stage-scheduling` for more details.
+
+The preset transpiler pipelines can also be configured at a high level by setting an
+``optimization_level``.  This is in integer from 0 to 3 inclusive, indicating the relative effort to
+put into attempting to optimize the circuit for the hardware.  Level 0 disables all unnecessary
+optimizations; only transformations needed to make the circuit runnable at all will be present.  On
+the other end, level 3 enables a full barrage of optimization techniques, some of which can be very
+expensive in compilation time.  Similar to classical compilers, optimization level 3 is not always
+guaranteed to produce the best results.  Qiskit defaults to optimization level 2.
```

Comment:
Should we explicitly argue why we chose level 2? I am thinking something like:

```suggestion
guaranteed to produce the best results.  Qiskit defaults to optimization level 2,
which offers a balanced tradeoff between compilation time and the expected level of circuit optimization.
```
I understand that the wording here can be tricky, so no need to apply the suggestion.

Reply:
Done in adacfe9ab7.